lib-find-meal/src/main/java/com/kinandcarta/lib/find/meal/view/FindMealActivity.kt

```diff
@@ -17,52 +27,148 @@ import kotlinx.coroutines.launch
 
 class FindMealActivity : AppCompatActivity() {
     private val mainScope = MainScope()
+    private val sdk = MealsSDK()
+    private val mealsAdapter = MealsAdapter(listOf())
 
     private lateinit var mealsRecyclerView: RecyclerView
     private lateinit var progressBarView: FrameLayout
     private lateinit var swipeRefreshLayout: SwipeRefreshLayout
 
-    private val sdk = MealsSDK()
-    private val mealsAdapter = MealsAdapter(listOf())
+    private lateinit var fusedLocationProviderClient: FusedLocationProviderClient
+    private lateinit var locationCallback: LocationCallback
+    private lateinit var locationRequest: LocationRequest
+    private var locationUpdateState = false
+    private lateinit var lastLocation: Location
 
     override fun onCreate(savedInstanceState: Bundle?) {
         super.onCreate(savedInstanceState)
 
-        title = "Find a meal"
-        setContentView(R.layout.find_meal_activity)
+        setContentView(com.kinandcarta.lib.find.meal.R.layout.find_meal_activity)
+        title = getString(R.string.find_meal_title) // TODO figure out how to reach strings from main res/strings
+
+        fusedLocationProviderClient = LocationServices.getFusedLocationProviderClient(this)
+        updateLastLocation()
 
-        mealsRecyclerView = findViewById(R.id.rvMeals)
-        progressBarView = findViewById(R.id.progressBar)
-        swipeRefreshLayout = findViewById(R.id.swipeContainer)
+        locationCallback = object : LocationCallback() {
+            override fun onLocationResult(p0: LocationResult?) {
+                super.onLocationResult(p0)
+
+                if (p0 != null) {
+                    lastLocation = p0.lastLocation
+                }
+            }
+        }
+        createLocationRequest()
+
+        mealsRecyclerView = findViewById(com.kinandcarta.lib.find.meal.R.id.rvMeals)
+        progressBarView = findViewById(com.kinandcarta.lib.find.meal.R.id.progressBar)
+        swipeRefreshLayout = findViewById(com.kinandcarta.lib.find.meal.R.id.swipeContainer)
 
         mealsRecyclerView.adapter = mealsAdapter
         mealsRecyclerView.layoutManager = LinearLayoutManager(this)
 
         swipeRefreshLayout.setOnRefreshListener {
             swipeRefreshLayout.isRefreshing = false
-            displayMeals(true)
+            displayMeals(false, true)
+        }
+        displayMeals(false, false)
+    }
+
+    private fun updateLastLocation() {
+        if (ActivityCompat.checkSelfPermission(this, Manifest.permission.ACCESS_FINE_LOCATION) == PERMISSION_GRANTED
+        ) {
+            fusedLocationProviderClient.lastLocation.addOnSuccessListener { location: Location ->
+                if (location != null) {
+                    lastLocation = location
```

Comment:
Same here RE the comment above about either using the elvis operator `?:` to get the location or just assign directly to lastLocation if lastLocation is nullable